### ISSUE TITLE
[Wolfi Package Request]: Make aws-cli package name default for v2

### ISSUE THREAD

**Issue by MPritsch** [wolfi-dev/os#37104]
### What software would you like us to add to wolfi-os. Ideally include a URL to the project and its source.

Hello, recently the default version of the package-name "aws-cli" was changed from v2 to v1. We installed the version in our docker-images using a pipeline and it broke a bunch of services, because the aws-cli v1 is not functional. I would argue that most users expect the default-package name to contain a recent version. v2 became the default aws-cli version since 2020. You would need some old unmaintained applications to still want to use v1.

Example how v1 ist broken:
```
/ # apk search aws-cli
aws-cli-1.32.71-r0
aws-cli-2-2.22.17-r0
aws-cli-v2-2.22.13-r0

/ # apk add aws-cli

/ # aws sts get-caller-identity
Traceback (most recent call last):
  File "/usr/bin/aws", line 19, in <module>
    import awscli.clidriver
ModuleNotFoundError: No module named 'awscli'
```

My request is that you put aws-cli v1 in it's own package like aws-cli-1 and move the v2 version back into the "aws-cli" package-name to be less error prone.

### Please check all that apply

- [ ] This package has an un-restrictive license
- [ ] The package/versions proposed are actively maintained upstream
- [ ] I am interested in adding this package to Wolfi OS myself
- [ ] I am willing to help maintain this package

**Comment by kranurag7** [comment]
@MPritsch thanks for opening the issue. 

We've made `v2` default in wolfi and that can be pulled using `apk add aws-cli-2` 

```yaml
  options:
    # melange generates incorrect SCA provides for vendored copies of
    # libraries
    no-provides: true
```
we can make it easier for you to add it simply using `apk add aws-cli` but we'll run into some bugs so not sure if I've a fix there. 
v1 is no longer available in wolfi because we only latest version in wolfi and as of writing this, the latest version of aws-cli is 2.

**Comment by MPritsch** [comment]
Yes, I saw that the package-name "aws-cli-2" contains the correct version. My issue is that I don't want or expect v1 to be under the package name "aws-cli". My expectation for the default package-name "aws-cli" as with any other package is that it contains the last stable release. In this case v2.

The reason why I opened this issue is that a week earlier, if I executed `apk add aws-cli`, I would get v2. Today I get v1. This is a breaking change as mentioned in my example.

I have similar problems with a bunch of other packages (e.g. argo-cd), which uses an old version (2.7.x) instead of the recent stable one (2.13.x).

If wolfi no longer maintains aws-cli v1, why is it the default version?

What kind of bug is there with aws-cli?

**Comment by kranurag7** [comment]
> if I executed `apk add aws-cli`, I would get v2. Today I get v1. This is a breaking change as mentioned in my example.

I accept and acknowledge that this is a breaking change but we've renamed this package [here](https://github.com/wolfi-dev/os/pull/36031) 
This is how we version stream all the other packages, this was long due. We don't include `v1` `v2` etc. in name. We just use `<package-name>-<numeric-package-version>` so that automation takes care of bumping it next time. 

`aws-cli` is available but is not updated anymore, we renamed the package and you'll have to explicitly pull it using `aws-cli-2`
Sorry for the inconvenience here. 


> I have similar problems with a bunch of other packages (e.g. argo-cd), which uses an old version (2.7.x) instead of the recent stable one (2.13.x).

Please open the issue when you see this occurring. 
```bash
/ # apk add argo-cd
(1/1) Installing argo-cd-2.13 (2.13.2-r1)
Executing busybox-1.37.0-r0.trigger
OK: 158 MiB in 16 packages
```
I checked it now and it's 2.13. I again want to stress on this that don't expect the version to be immediately available after the release. Automation takes some time, please check open PR queues for that. In 99% cases, you'll find that the version is latest if you'll give the automation 24-48 hrs, sometimes more if there's some build fixes that needs to be done. 




> If wolfi no longer maintains aws-cli v1, why is it the default version?

the latest is `aws-cli-2` please be explicit while adding using `apk add` 



> What kind of bug is there with aws-cli?

I don't know as of now but I can track it down and get back to you. I acknowledge your point that we should have a `provides` for this which enables you to get v2 when you do `apk add aws-cli`

**Comment by MPritsch** [comment]
I'm torn on this. I understand that you want to people to be explicit in their versioning. However this makes it hard to automate because you need to keep track of changing version numbers and changing package-names. I do not like this design decision.

It wouldn't be an issue if you allow versions to be selected by the base-package-name. E.g. select the v2 version:
```
/ # apk add aws-cli=2.22.17-r0
ERROR: unable to select packages:
  aws-cli-1.32.71-r0:
    breaks: world[aws-cli=2.22.17-r0]
```

This change with aws-cli breaks my trust in this package-manager. A package-name should never roll back to an older version unless there is a security issue or major breaking change of some kind.

**Comment by EyeCantCU** [comment]
Hi @MPritsch, thank you for raising! We fully agree with you here. We just landed a couple changes that will fix this, but it may take a few moments to take affect. The latest version of AWS CLI will always provide `aws-cli` going forward

**Comment by kranurag7** [comment]
@MPritsch sorry for the breaking change here. 
we've added provides in config and now if you do `apk add aws-cli` you'll get v2 which is expected. 

It is being done here https://github.com/wolfi-dev/os/pull/37133 and should be good to go in 15 minutes. 
Appreciate you opening the issue and please do that when you notice a a version mismatch.

**Comment by MPritsch** [comment]
Thanks for changing the behavior `aws-cli` to provide v2. I will check tomorrow if the index is updated and then close the ticket.
Best regards

**Comment by MPritsch** [comment]
Hello,
I checked the apkindex and the installation looks good. However I'm a bit confused as to why the apk search for "aws-cli" displays the version 1.32.71-r0 when instead 2.22.21-r0 is installed.

```
/ # apk update
...
/ # apk search aws-cli
aws-cli-1.32.71-r0
aws-cli-2-2.22.21-r0
aws-cli-v2-2.22.13-r0
/ # apk add aws-cli
(1/5) Installing libbz2-1 (1.0.8-r9)
(2/5) Installing perl (5.40.0-r3)
(3/5) Installing libstdc++ (14.2.0-r6)
(4/5) Installing groff (1.23.0-r3)
(5/5) Installing aws-cli-2 (2.22.21-r0)
Executing glibc-2.40-r3.trigger
Executing busybox-1.37.0-r0.trigger
OK: 237 MiB in 20 packages
/ # aws --version
aws-cli/2.22.21 Python/3.11.11 Linux/5.15.0-127-generic source-exe/aarch64.wolfi.20230201
```

Thanks anyway, this looks very good so far!

**Comment by kranurag7** [comment]
@MPritsch `apk search` display all the occurrences of the query you searched for and if that's available in the index. We've both as of now for aws-cli and for some other packages as well but we intend to keep only the latest version in wolfi.

**Comment by cmwilson21** [comment]
Thanks for the contribution!

We’re closing this as part of a repo-wide reset to support upcoming changes in how contributions are validated and merged. You can find more context in our discussion post here: [Upcoming changes to wolfi-dev/os](https://github.com/orgs/wolfi-dev/discussions/77550).

If you’d like to continue with this work, please feel free to reopen or resubmit your issue or PR after reviewing the updated [Contribution Guidelines](https://github.com/wolfi-dev/os/blob/main/CONTRIBUTING.md).

We appreciate your understanding and your continued contributions to Wolfi.
